### PR TITLE
use pathlib in adaptative.py

### LINKED ISSUE #872: Make IVADOMED Python API compatible with Path() objects (from pathlib)
### Motivation for the feature
Provides better compatibility across platforms, and is the more modern way to handle paths/files/suffixes/etc

https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/

### Description of the feature
Make it so the IVADOMED API (e.g. https://github.com/ivadomed/ivadomed/blob/7a595ae39e94085837582f7176b8ec59bd9a2bc9/ivadomed/inference.py#L311 path to model and image) accepts objects of the class `from pathlib import Path` type.

### PR DESCRIPTION
**Pull request by cakester**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [ ] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Linked issues

resolves #872

### CODE REVIEW

File: requirements_common.txt [hunk 1]
```diff
@@ -6,11 +6,12 @@ matplotlib>=3.3.0
 nibabel~=3.2
 onnxruntime==1.4.0
 pandas~=1.1
+pathlib~=1.0.1
```
Comment (dyt811) [inline requirements_common.txt:9]: Isn't pathlib native to Python at this point? Do you still need to version that? Just curious.

File: ivadomed/loader/adaptative.py [hunk 2]
```diff
@@ -751,7 +750,7 @@ def HDF5ToBIDS(path_hdf5, subjects, path_dir):
                 for filename in grp['gt/{}'.format(ct)].attrs['gt_filenames']:
                     gt_data = grp['gt/{}'.format(ct)]
                     nib_image = nib.Nifti1Image(gt_data, np.eye(4))
-                    filename = os.path.join(path_label, filename.split("/")[-1])
+                    filename = Path(path_label).joinpath(filename.split("/")[-1])
```
Comment (dyt811) [inline ivadomed/loader/adaptative.py:753]: Can probably  just use `/` for simpler readability when only a single string.

File: ivadomed/loader/adaptative.py [hunk 3]
```diff
@@ -715,19 +714,19 @@ def HDF5ToBIDS(path_hdf5, subjects, path_dir):
...
 
         # loop over all subjects
         for sub in subjects:
             path_sub = path_dir + '/' + sub + '/anat/'
             path_label = path_dir + '/derivatives/labels/' + sub + '/anat/'
 
-            if not path.exists(path_sub):
-                os.makedirs(path_sub)
+            if not Path(path_sub).exists():
+                Path(path_sub).mkdir(parents=True)
```
Comment (dyt811) [inline ivadomed/loader/adaptative.py:726]: 👍

**Review by dyt811 — APPROVED** [review]

File: requirements_common.txt (hunk 1 above)
Comment (cakester) [reply requirements_common.txt:9]: you are right, i missed that

**Pull request merged**